VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,184 @@
+package org.edx.mobile.discussions;
+
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ClickableSpan;
+import android.view.View;
+import android.widget.TextView;
+
+import org.edx.mobile.R;
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+public class DiscussionTextUtilsTest extends BaseTestCase {
+
+    private TextView dummyTextView;
+    private Runnable dummyClickListener;
+    private String endorsePrefix;
+    private String answerPrefix;
+    private long now = System.currentTimeMillis();
+
+    @Before
+    public void initValues() {
+        dummyTextView = new TextView(context);
+        dummyClickListener = Mockito.mock(Runnable.class);
+        endorsePrefix = context.getString(R.string.discussion_post_endorsed);
+        answerPrefix = context.getString(R.string.discussion_post_marked_as_answer);
```

Comment:
I think it was fine as it was. These string format is unlikely to change without also changing the implementations, and if the actual string is changed then the hardcoded implementation is more likely to get out of sync and fail.